src/test/java/io/openliberty/tools/intellij/it/SingleModMPProjectTestCommon.java

```diff
@@ -986,6 +991,35 @@ public void testCustomStartParametersClearedOnConfigRemoval() {
         }
     }
 
+    /**
+     * Deletes the directory specified by dirPath if it exists.
+     *
+     * @param dirPath The path to the directory that may be deleted.
+     */
+    private static void deleteDirectoryIfExists(String dirPath) {
+        File dir = new File(dirPath);
+        if (dir.exists()) {
+            TestUtils.deleteDirectory(dir);
+        }
+    }
+
+    /**
+     * Returns the gradle project directory path with space in directory name
+     *
+     * @return The projects directory path.
+     */
+    private static String getGradleProjectSIDDirPath() {
```

Comment:
There is already methods to get the directory path 'getProjectsDirPath'. Then why do we need this and getMavenProjectSIDDirPath method?

Reply:
`getProjectsDirPath()` is an abstract method, and we can't use it in the `cleanup()` method, which is annotated with `@ AfterAll` . 
We only need to call the method `deleteDirectoryIfExists()` if the path string is:

```
Paths.get("src", "test", "resources", "projects", "gradle sample").toAbsolutePath().toString();
Paths.get("src", "test", "resources", "projects", "maven sample").toAbsolutePath().toString();
```
We don't need to call the method `deleteDirectoryIfExists()` if the path string is:


```
Paths.get("src", "test", "resources", "projects", "gradle").toAbsolutePath().toString();
Paths.get("src", "test", "resources", "projects", "maven").toAbsolutePath().toString();
```
Since we are using an inheritance approach, do we need to change the following method to non-abstract?


`public abstract String getProjectsDirPath();`

Reply:
Why can't we write the after all method inside GradleSingleModMPSIDProjectTest and MavenSingleModMPSIDProjectTest?

    @AfterAll
    public static void cleanup() {
        deleteDirectoryIfExists(PROJECTS_PATH_NEW);
    }